src/cache/cache-manager.ts

```diff
@@ -77,31 +69,36 @@ export class CacheManager {
     await this.keyManifest?.add(cacheKey.toKey());
   }
 
-  async clear(): Promise<void> {
+  async clear(clientId?: string): Promise<void> {
     const keys = await this.getCacheKeys();
 
     /* istanbul ignore next */
     if (!keys) return;
 
-    keys.forEach(async key => {
-      await this.cache.remove(key);
-    });
+    await keys
+      .filter(key => (clientId ? key.includes(clientId) : true))
+      .reduce(async (memo, key) => {
+        await memo;
+        await this.cache.remove(key);
+      }, Promise.resolve());
```

Comment:
Previously these removals were not being awaited properly, this reducer ensures that deletions are happening in sequence.